### ISSUE TITLE
Using electron-builder-notarize-pkg in an action

### ISSUE THREAD

**Issue by gungorbudak** [karaggeorge/electron-builder-notarize-pkg#3]
Hi,

I'm using https://github.com/samuelmeuli/action-electron-builder as the action to build my app with `pkg` installer and I'm signing and notarizing both the app and the installer.

I exported an application and an installer certificate in a single `p12` file and then set an ENV variable as the action requires.

I have `"afterSign": "electron-builder-notarize"` and `"afterAllArtifactBuild": "electron-builder-notarize-pkg"`.

I also provide an API key file, API_KEY_ID and API_KEY_ISSUER_ID and PSC_NAME ENV variables as this package requires in the action.

I tested this locally and both app and installer were signed and notarized successfully. However, I couldn't sign the installer with the action on GitHub. I provide everything as secret and then provide ENV variables in the action.

```
Building the Electron app…
yarn run v1.22.4
$ /Users/runner/work/path/node_modules/.bin/electron-builder --mac -c.publish.provider=generic -c.publish.url=url
  • electron-builder  version=22.6.0 os=19.6.0
  • artifacts will be published if draft release exists  reason=CI detected
  • loaded configuration  file=package.json ("build" field)
  • loaded parent configuration  file=/Users/runner/work/path/node_modules/electron-webpack/out/electron-builder.js
  • packaging       platform=darwin arch=x64 electron=8.2.5 appOutDir=dist/mac
  • downloading     url=https://github.com/electron/electron/releases/download/v8.2.5/electron-v8.2.5-darwin-x64.zip size=66 MB parts=8
  • downloaded      url=https://github.com/electron/electron/releases/download/v8.2.5/electron-v8.2.5-darwin-x64.zip duration=938ms
  • signing         file=dist/mac/app.app identityName=Developer ID Application: Name (ID) identityHash=6A0830E69974A4AD816360C951A70CECCAD26C2A provisioningProfile=none
Notarizing com.company.app found at /Users/runner/work/path/dist/mac/app.app
Done notarizing com.company.app
  • building        target=pkg arch=x64 file=dist/setup-app.pkg
Signing /Users/runner/work/path/dist/setup-app.pkg...
  ⨯ Failed to sign /Users/runner/work/path/dist/setup-app.pkg

Command failed with exit code 1: productsign --sign *** /Users/runner/work/path/dist/setup-app.pkg /Users/runner/work/path/dist/setup-app-signed.pkg
productsign: error: Could not find appropriate signing identity for “***”.  stackTrace=
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Error: Failed to sign /Users/runner/work/path/dist/setup-app.pkg
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Command failed with exit code 1: productsign --sign *** /Users/runner/work/path/dist/app.pkg /Users/runner/work/path/dist/setup-app-signed.pkg
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           productsign: error: Could not find appropriate signing identity for “***”.
```
The reason `PSC_NAME` is looking as `***` is that it was given as a secret and GitHub hides it in the logs. It looks like this: `Developer ID Installer: Name (ID)`.

Do you have any idea how I can make this work in the action? It looks like locally the package can find the certificate through Keychain but in the action somehow it is not available although electron-builder can access the certificate through its ENV variable `CSC_LINK`.

Thanks much in advance.

Edit:

The part of the workflow that does the building:

```
      - name: Prepare for app notarization (macOS)
        if: startsWith(matrix.os, 'macos')
        # Import Apple API key for app notarization on macOS
        run: |
          mkdir -p ~/private_keys/
          echo '${{ secrets.MAC_API_KEY }}' > ~/private_keys/AuthKey_${{ secrets.MAC_API_KEY_ID }}.p8

      - name: Build and release app
        uses: samuelmeuli/action-electron-builder@v1
        with:
          build_script_name: "compile-test"
          args: "-c.publish.provider=generic -c.publish.url=url"
          # GitHub token, automatically provided to the action
          # (No need to define this secret in the repo settings)
          github_token: ${{ secrets.github_token }}

          # macOS code signing certificate
          mac_certs: ${{ secrets.MAC_CERTS }}
          mac_certs_password: ${{ secrets.MAC_CERTS_PASSWORD }}

          # If the commit is tagged with a version (e.g. "v1.0.0"),
          # release the app after building
          # release: ${{ startsWith(github.ref, 'refs/tags/test-v') }}
        env:
          # macOS notarization API key
          API_KEY_ID: ${{ secrets.MAC_API_KEY_ID }}
          API_KEY_ISSUER_ID: ${{ secrets.MAC_API_KEY_ISSUER_ID }}
          # macOS installer notarization requirements
          PSC_NAME: ${{ secrets.MAC_PSC_NAME }}
```

**Comment by karaggeorge** [comment]
Hey, I had mimicked the code sign process from electron-builder. I guess I might have missed something. I'll take a look and try to figure out where

**Comment by gungorbudak** [comment]
Hi @karaggeorge did you have a chance to look at this? I was testing locally by deleting certificates from Keychain Access and specifying base64 encoded certificates with `CSC_LINK` and it seems `electron-builder` creates a temporary keychain for signing and somehow I suspect it gets deleted before `electron-builder-notarize-pkg` and I'm still unable to solve it.

**Comment by gungorbudak** [comment]
Dear @karaggeorge, after I set `DEBUG=electron-builder`, I could obtain more information and it seems `electron-builder` deletes the temporary keychain before this package so that's why it doesn't find the certificate.

However, I also saw that `electron-builder` does signing for `.pkg` as well please check the below log. I managed to notarize the installer by disabling signing with this package. I checked the resulting installer for signature and notarization and both are validated.

Do you think it makes sense to remove signing with this package and keep everything else?

```
  • executed        file=pkgbuild stdout=
                      pkgbuild: Reading components from /Users/gungor/Desktop/Projects/Personal/path/dist/com.company.app.plist
                      pkgbuild: Adding component at app.app
                      pkgbuild: Wrote package to /Users/gungor/Desktop/Projects/Personal/path/dist/com.company.app.pkg
                      
  • executing       file=productbuild args=--sign HASH --keychain /var/folders/pt/bpdgv2t918g7pxjd58m57nc80000gn/T/db3cdf0da6728a957a5c314b12ad4f1cc32c7ccb19c370ef2d9c6bb0b53ff4bc.keychain --distribution /Users/gungor/Desktop/Projects/Personal/path/dist/distribution.xml /Users/gungor/Desktop/Projects/Personal/path/dist/setup-app.pkg cwd=/Users/gungor/Desktop/Projects/Personal/path/dist
  • executed        file=productbuild stdout=
                      productbuild: Using timestamp authority for signature
                      productbuild: Signing product with identity "Developer ID Installer: Name (ID)" from keychain /private/var/folders/pt/bpdgv2t918g7pxjd58m57nc80000gn/T/db3cdf0da6728a957a5c314b12ad4f1cc32c7ccb19c370ef2d9c6bb0b53ff4bc.keychain
                      productbuild: Adding certificate "Developer ID Certification Authority"
                      productbuild: Adding certificate "Apple Root CA"
                      productbuild: Wrote product to /Users/gungor/Desktop/Projects/Personal/path/dist/setup-app.pkg
                      
  • executing       file=security args=delete-keychain /var/folders/pt/bpdgv2t918g7pxjd58m57nc80000gn/T/db3cdf0da6728a957a5c314b12ad4f1cc32c7ccb19c370ef2d9c6bb0b53ff4bc.keychain
  • executed        file=security
Notarizing /Users/gungor/Desktop/Projects/Personal/path/dist/setup-app.pkg...
Notarized /Users/gungor/Desktop/Projects/Personal/path/dist/setup-app.pkg successfully.
```

**Comment by karaggeorge** [comment]
That actually solves the issue then. I don't think electron-builder used to sign the pkg file when I created this, but if they do now, I can remove that whole logic and just notarize it. I'll push a new build and post here. If you're able to test that'd be awesome

**Comment by karaggeorge** [comment]
Published as `electron-builder-notarize-pkg@1.2.0` with removing the signing part and just keeping the notarizing logic. Can you check?

**Comment by gungorbudak** [comment]
Hey @karaggeorge I tested the new version, works great! Thanks much
